kong/api/routes/upstreams.lua

```diff
@@ -40,7 +42,48 @@ return {
     end,
 
     GET = function(self, dao_factory)
-      crud.paginated_set(self, dao_factory.targets)
+      -- by default, just get a paginated list
+      if not self.params.active then
+        return crud.paginated_set(self, dao_factory.targets)
+      end
+
+      self.params.active = nil
```

Comment:
My thought here was that `active` is a modifier of the resource, not a separate (or sub) resource. This is why I choose to leave it as a query param, not a URI. But my experience defining RESTful routes is very limited :) Let me know what your thoughts are given the above, I will happily defer to your expertise here